libcontainer/init_linux.go

```diff
@@ -440,35 +440,62 @@ func syncParentSeccomp(pipe *os.File, seccompFd *os.File) error {
 	return readSync(pipe, procSeccompDone)
 }
 
-// setupUser changes the groups, gid, and uid for the user inside the container
-func setupUser(config *initConfig) error {
-	// Set up defaults.
-	defaultExecUser := user.ExecUser{
-		Uid:  0,
-		Gid:  0,
-		Home: "/",
-	}
-
-	passwdPath, err := user.GetPasswdPath()
+func getExecUser(ustr string) (*user.ExecUser, error) {
+	var uid, gid int
+	u := strings.SplitN(ustr, ":", 2)
```

Comment:
I have yet to look at the rest of the changes, but we can use `strings.Cut` here;


```go
user, group, ok := strings.Cut(ustr, ":")
```

That would return `ok` if there was a colon (so checking `len()` is not needed, and directly assign the results to ready-to-use variables.

Reply:
Yes, I was confused by the code in `user.GetExecUserPath`, in `config.json`, gid is alway exist, so I should not deal with the condition of `gid` is not provided.